messages/src/msg_fields/protocols/pickup/delivery.rs

```diff
@@ -0,0 +1,31 @@
+use serde::{Deserialize, Serialize};
+use serde_with::{base64::Base64, serde_as};
+use typed_builder::TypedBuilder;
+
+use super::decorators::PickupDecoratorsCommon;
+use crate::msg_parts::MsgParts;
+
+pub type Delivery = MsgParts<DeliveryContent, PickupDecoratorsCommon>;
+
+#[derive(Clone, Debug, Deserialize, Serialize, Default, PartialEq, TypedBuilder)]
+pub struct DeliveryContent {
+    #[builder(default, setter(strip_option))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub recipient_key: Option<String>,
+    #[serde(rename = "~attach")]
+    pub attach: Vec<DeliveryAttach>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, Default, PartialEq, TypedBuilder)]
+pub struct DeliveryAttach {
+    #[serde(rename = "@id")]
+    pub id: String,
+    pub data: DeliveryAttachData,
+}
+
+#[serde_as]
+#[derive(Clone, Debug, Deserialize, Serialize, Default, PartialEq, TypedBuilder)]
+pub struct DeliveryAttachData {
+    #[serde_as(as = "Base64")]
+    pub base64: Vec<u8>,
```

Comment:
did you consider reusing the `Attachment` type already within the `messages` crate?

Reply:
I wasn't aware of `Attachment` type. Thanks for bringing it to my notice!

Reply:
I have now taken a look at the Attachment type. From reading the pickup-v2 RFC, it is unclear to me if delivery messages contain general attachments or only those with base64 formatted data. I have [sought clarification](https://github.com/hyperledger/aries-rfcs/issues/797).

Another difference is the necessity of "@id" field in pickup (for later marking as received), while it is optional in general attachment. 

For now, I'd like to keep it as a separate structure, but I shall try to align it with the already defined type. Later they can be merged if appropriate. 

Does that sound fine?

Reply:
Ahh, thanks for looking into that. In that case, I'm happy with it how it is, so long as the other reviewers are fine with it too 👍

Reply:
Done.